### PR TITLE
feat: Support custom artifact notice

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Add new `artifactNotice` field that could be optionally returned with successful file modification

## Context

- Ref: #27775

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/update/branch/get-updated.spec.ts [hunk 1]
```diff
@@ -28,6 +30,7 @@ jest.mock('../../../../modules/manager/composer');
 jest.mock('../../../../modules/manager/helmv3');
 jest.mock('../../../../modules/manager/npm');
 jest.mock('../../../../modules/manager/git-submodules');
+jest.mock('../../../../modules/manager/gomod');
```
Comment (viceice) [inline lib/workers/repository/update/branch/get-updated.spec.ts:33]: can't we simply use one of the existing managers? they are all mocked anyways 


```suggestion
jest.mock('../../../../modules/manager/gomod', () => mockDeep ());
```
perf, we should do it for all mocks

**Comment by rarkins** [comment]
Reproduction link?

**Comment by zharinov** [comment]
> Reproduction link?

Well, the same as for gomod: https://github.com/renovate-testing/test-27775-go-get-multiple-deps/pull/1

Can't provide it separately, without particular caller implementation

File: lib/workers/repository/update/branch/index.ts [hunk 2]
```diff
@@ -147,6 +147,7 @@ export async function processBranch(
   }
   const keepUpdatedLabel = config.keepUpdatedLabel;
   const artifactErrorTopic = emojify(':warning: Artifact update problem');
+  const artifactNoticeTopic = emojify(':warning: Notice');
```
Comment (rarkins) [inline lib/workers/repository/update/branch/index.ts:150]:
```suggestion
  const artifactNoticeTopic = emojify(':warning: Artifact update notice');
```

File: lib/workers/repository/update/branch/index.ts [hunk 3]
```diff
@@ -147,6 +147,7 @@ export async function processBranch(
   }
   const keepUpdatedLabel = config.keepUpdatedLabel;
   const artifactErrorTopic = emojify(':warning: Artifact update problem');
+  const artifactNoticeTopic = emojify(':warning: Artifact update notice');
```
Comment (rarkins) [inline lib/workers/repository/update/branch/index.ts:150]:
```suggestion
  const artifactNoticeTopic = emojify(':information_source: Artifact update notice');
```

Please check how this one looks

**Comment by rarkins** [comment]
# :information_source: Artifact update notice

**Comment by zharinov** [comment]
Could we merge this one?

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**